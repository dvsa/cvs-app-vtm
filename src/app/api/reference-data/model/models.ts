export * from './reference-data.model';
