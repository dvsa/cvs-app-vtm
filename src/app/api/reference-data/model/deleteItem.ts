/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}