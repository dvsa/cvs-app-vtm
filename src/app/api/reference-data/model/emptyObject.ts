/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
