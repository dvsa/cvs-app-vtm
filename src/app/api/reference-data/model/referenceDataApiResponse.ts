/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */
import { ReferenceDataApiResponseWithPagination } from './referenceDataApiResponseWithPagination';
import { ReferenceDataApiResponseWithoutPagination } from './referenceDataApiResponseWithoutPagination';

export type ReferenceDataApiResponse = ReferenceDataApiResponseWithoutPagination | ReferenceDataApiResponseWithPagination;