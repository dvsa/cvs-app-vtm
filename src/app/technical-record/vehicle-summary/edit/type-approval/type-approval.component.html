<ng-container>
  <div class="govuk-form-group">
    <fieldset class="govuk-fieldset">
      <h2 class="govuk-heading-m govuk-!-margin-bottom-2">Type approval</h2>
      <form [formGroup]="techRecordFg">
        <!-- Approval type -->
        <div class="govuk-form-group">
          <label for="approvalType" class="govuk-label govuk-!-font-weight-bold">
            Approval type
          </label>
          <select id="approvalType" formControlName="approvalType" class="govuk-select">
            <option *ngFor="let type of approvalTypeOptions" [ngValue]="type.name">
              {{ type.name }}
            </option>
          </select>
        </div>

        <!-- Approval type number -->
        <div class="govuk-form-group">
          <label for="approvalTypeNumber" class="govuk-label govuk-!-font-weight-bold">
            Approval type number (optional)
          </label>
          <input
            id="approvalTypeNumber"
            type="text"
            name="approvalTypeNumber"
            maxlength="25"
            formControlName="approvalTypeNumber"
            class="govuk-input govuk-input govuk-input--width-20"
          />
        </div>

        <!-- National type approval -->
        <div class="govuk-form-group">
          <label for="ntaNumber" class="govuk-label govuk-!-font-weight-bold">
            National type approval number (optional)
          </label>
          <input
            id="ntaNumber"
            type="text"
            name="ntaNumber"
            maxlength="40"
            formControlName="ntaNumber"
            class="govuk-input govuk-input govuk-input--width-20"
          />
        </div>

        <!-- Variant number -->
        <div class="govuk-form-group">
          <label for="variantNumber" class="govuk-label govuk-!-font-weight-bold">
            Variant number (optional)
          </label>
          <input
            id="variantNumber"
            type="text"
            name="variantNumber"
            maxlength="25"
            formControlName="variantNumber"
            class="govuk-input govuk-input govuk-input--width-20"
          />
        </div>

        <!-- Variant version number -->
        <div class="govuk-form-group">
          <label for="variantVersionNumber" class="govuk-label govuk-!-font-weight-bold">
            Variant version number (optional)
          </label>
          <input
            id="variantVersionNumber"
            type="text"
            name="variantVersionNumber"
            maxlength="35"
            formControlName="variantVersionNumber"
            class="govuk-input govuk-input govuk-input--width-20"
          />
        </div>
      </form>
    </fieldset>
  </div>
</ng-container>
