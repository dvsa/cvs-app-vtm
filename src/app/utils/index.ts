export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


