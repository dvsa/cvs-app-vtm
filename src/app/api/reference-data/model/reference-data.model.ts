import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
