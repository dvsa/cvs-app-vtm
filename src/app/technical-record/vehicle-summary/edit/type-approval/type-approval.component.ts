import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { ControlContainer, FormGroupDirective, FormGroup, FormBuilder } from '@angular/forms';

import { TechRecord } from '@app/models/tech-record.model';
import { SelectOption } from '@app/models/select-option';
import { DisplayOptionsPipe } from '@app/pipes/display-options.pipe';
import { APPROVAL_TYPE } from '@app/technical-record/technical-record.constants';

@Component({
  selector: 'vtm-type-approval',
  templateUrl: './type-approval.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
  viewProviders: [
    {
      provide: ControlContainer,
      useExisting: FormGroupDirective
    }
  ]
})
export class TypeApprovalComponent implements OnInit {
  @Input() activeRecord: TechRecord;

  techRecordFg: FormGroup;
  approvalTypeOptions: SelectOption[];

  constructor(private parent: FormGroupDirective, private fb: FormBuilder) {}

  ngOnInit() {
    this.approvalTypeOptions = new DisplayOptionsPipe().transform(APPROVAL_TYPE);

    this.techRecordFg = this.parent.form as FormGroup;
    this.techRecordFg.addControl('approvalType', this.fb.control(this.activeRecord.approvalType));
    this.techRecordFg.addControl(
      'approvalTypeNumber',
      this.fb.control(this.activeRecord.approvalTypeNumber)
    );
    this.techRecordFg.addControl('ntaNumber', this.fb.control(this.activeRecord.ntaNumber));
    this.techRecordFg.addControl(
      'variantNumber',
      this.fb.control(this.activeRecord.variantNumber)
    );
    this.techRecordFg.addControl(
      'variantVersionNumber',
      this.fb.control(this.activeRecord.variantVersionNumber)
    );
  }
}
