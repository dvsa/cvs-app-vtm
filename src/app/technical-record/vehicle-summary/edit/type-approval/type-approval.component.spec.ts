import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {
  ReactiveFormsModule,
  FormsModule,
  FormGroupDirective,
  FormGroup,
  AbstractControl
} from '@angular/forms';

import { TESTING_UTILS } from '@app/utils/testing.utils';
import { TypeApprovalComponent } from './type-approval.component';
import { TechRecord } from '@app/models/tech-record.model';

const mockTechRecord = {
  approvalType: 'NTA',
  approvalTypeNumber: 'approNum',
  variantNumber: 'varNum',
  variantVersionNumber: 'varVerNum'
} as TechRecord;

describe('TypeApprovalComponent', () => {
  let component: TypeApprovalComponent;
  let fixture: ComponentFixture<TypeApprovalComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule],
      declarations: [TypeApprovalComponent],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective({
            techRecord: new FormGroup({}) as AbstractControl
          })
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TypeApprovalComponent);
    component = fixture.componentInstance;
  });

  it('should create with empty control states', () => {
    component.techRecord = {} as TechRecord;
    fixture.detectChanges();

    expect(component).toBeDefined();
    expect(component.techRecordFg.get('approvalType').value).toEqual(null);
    expect(component.techRecordFg.get('approvalTypeNumber').value).toEqual(null);
    expect(component.techRecordFg.get('variantNumber').value).toEqual(null);
    expect(component.techRecordFg.get('variantVersionNumber').value).toEqual(null);

    expect(fixture).toMatchSnapshot();
  });

  it('should create with initialized form controls', () => {
    component.techRecord = mockTechRecord;
    fixture.detectChanges();

    expect(component.techRecordFg.get('approvalType').value).toEqual(mockTechRecord.approvalType);
    expect(component.techRecordFg.get('approvalTypeNumber').value).toEqual(
      mockTechRecord.approvalTypeNumber
    );
    expect(component.techRecordFg.get('variantNumber').value).toEqual(
      mockTechRecord.variantNumber
    );
    expect(component.techRecordFg.get('variantVersionNumber').value).toEqual(
      mockTechRecord.variantVersionNumber
    );
  });
});
