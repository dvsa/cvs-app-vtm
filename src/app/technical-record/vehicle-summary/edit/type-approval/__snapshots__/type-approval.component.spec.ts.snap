// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeApprovalComponent should create with empty control states 1`] = `
<vtm-type-approval
  approvalTypeOptions={[Function Array]}
  fb={[Function FormBuilder]}
  parent={[Function FormGroupDirective]}
  techRecord={[Function Object]}
  techRecordFg={[Function FormGroup]}
>
  <div
    class="govuk-form-group"
  >
    <fieldset
      class="govuk-fieldset"
    >
      <h2
        class="govuk-heading-m govuk-!-margin-bottom-2"
      >
        Type approval
      </h2>
      <form
        class="ng-untouched ng-pristine ng-valid"
        ng-reflect-form="[object Object]"
        novalidate=""
      >
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label govuk-!-font-weight-bold"
            for="test-edit-approvalType"
          >
             Approval type 
          </label>
          <select
            class="govuk-select ng-untouched ng-pristine ng-valid"
            formcontrolname="approvalType"
            id="test-edit-approvalType"
            ng-reflect-name="approvalType"
          >
            
            <option
              ng-reflect-ng-value="NTA"
              value="0: NTA"
            >
               NTA 
            </option>
            <option
              ng-reflect-ng-value="ECTA"
              value="1: ECTA"
            >
               ECTA 
            </option>
            <option
              ng-reflect-ng-value="IVA"
              value="2: IVA"
            >
               IVA 
            </option>
            <option
              ng-reflect-ng-value="NSSTA"
              value="3: NSSTA"
            >
               NSSTA 
            </option>
            <option
              ng-reflect-ng-value="ECSSTA"
              value="4: ECSSTA"
            >
               ECSSTA 
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label govuk-!-font-weight-bold"
            for="test-edit-approvalTypeNumber"
          >
             Approval type number (optional) 
          </label>
          <input
            class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
            formcontrolname="approvalTypeNumber"
            id="test-edit-approvalTypeNumber"
            maxlength="25"
            name="approvalTypeNumber"
            ng-reflect-maxlength="25"
            ng-reflect-name="approvalTypeNumber"
            type="text"
          />
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label govuk-!-font-weight-bold"
            for="test-edit-ntaNumber"
          >
             National type approval number (optional) 
          </label>
          <input
            class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
            formcontrolname="ntaNumber"
            id="test-edit-ntaNumber"
            maxlength="40"
            name="ntaNumber"
            ng-reflect-maxlength="40"
            ng-reflect-name="ntaNumber"
            type="text"
          />
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label govuk-!-font-weight-bold"
            for="test-edit-variantNumber"
          >
             Variant number (optional) 
          </label>
          <input
            class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
            formcontrolname="variantNumber"
            id="test-edit-variantNumber"
            maxlength="25"
            name="variantNumber"
            ng-reflect-maxlength="25"
            ng-reflect-name="variantNumber"
            type="text"
          />
        </div>
        <div
          class="govuk-form-group"
        >
          <label
            class="govuk-label govuk-!-font-weight-bold"
            for="test-edit-variantVersionNumber"
          >
             Variant version number (optional) 
          </label>
          <input
            class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
            formcontrolname="variantVersionNumber"
            id="test-edit-variantVersionNumber"
            maxlength="35"
            name="variantVersionNumber"
            ng-reflect-maxlength="35"
            ng-reflect-name="variantVersionNumber"
            type="text"
          />
        </div>
      </form>
    </fieldset>
  </div>
</vtm-type-approval>
`;
